app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostActivity.java

```diff
@@ -0,0 +1,555 @@
+              // and check the result in onActivityResult().
+              resolvable.startResolutionForResult(
+                  ZimHostActivity.this,
+                  LOCATION_SETTINGS_PERMISSION_RESULT);
+            } catch (IntentSender.SendIntentException e) {
+              // Ignore the error.
+            } catch (ClassCastException e) {
+              // Ignore, should be an impossible error.
+            }
+            break;
+          case LocationSettingsStatusCodes.SETTINGS_CHANGE_UNAVAILABLE:
+            // Location settings are not satisfied. However, we have no way to fix the
+            // settings so we won't show the dialog.
+            break;
+          default:
+            break;
+        }
+      }
+    });
+  }
+
+  //Advice user to turn on hotspot manually for API<26
+  private void startHotspotManuallyDialog() {
+    AlertDialog.Builder builder = new AlertDialog.Builder(this, dialogStyle());
+
+    builder.setPositiveButton(getString(R.string.go_to_settings_label),
+        (dialog, id) -> launchTetheringSettingsScreen());
+
+    builder.setNeutralButton(getString(R.string.hotspot_dialog_neutral_button), (dialog, id) -> {
+
+      progressDialog =
+          ProgressDialog.show(this, getString(R.string.progress_dialog_starting_server), "",
+              true);
+      progressDialog.show();
+
+      startFlowable();
+    });
+
+    builder.setTitle(getString(R.string.hotspot_dialog_title));
+    builder.setMessage(
+        getString(R.string.hotspot_dialog_message)
+    );
+    AlertDialog dialog = builder.create();
+    dialog.show();
+  }
+
+  //Keeps checking if hotspot has been turned using the ip address with an interval of 1 sec
+  //If no ip is found after 15 seconds, dismisses the progress dialog
+  private void startFlowable() {
+    Flowable.fromCallable(() -> ServerUtils.getIp())
+        .retryWhen(error -> error.delay(1, TimeUnit.SECONDS))
+        .timeout(15, TimeUnit.SECONDS)
+        .firstOrError()
+        .observeOn(AndroidSchedulers.mainThread())
+        .subscribe(new SingleObserver<String>() {
+          @Override public void onSubscribe(Disposable d) {
+          }
+
+          @Override public void onSuccess(String s) {
+            progressDialog.dismiss();
+            serviceIntent.putStringArrayListExtra(SELECTED_ZIM_PATHS_KEY, selectedBooksPath);
+            serviceIntent.setAction(ACTION_START_SERVER);
+            startService(serviceIntent);
+            Log.d(TAG, "onSuccess:  " + s);
+          }
+
+          @Override public void onError(Throwable e) {
+            // display the ip and don't forget to dismiss the dialog
+            progressDialog.dismiss();
+            Toast.makeText(ZimHostActivity.this, R.string.server_failed_message, Toast.LENGTH_SHORT)
+                .show();
+            Log.d(TAG, "Unable to turn on server", e);
+          }
+        });
+  }
+
+  void createHotspotIntent(String ACTION) {
+    serviceIntent.setAction(ACTION);
```

Comment:
the `create*Intent` methods should return intent objects, not call startService

Reply:
Fixed. 6cafde506199b2ed4c07773edecb50345cb26409